python/astro_metadata_translator/translators/sdss.py

```diff
@@ -167,22 +172,27 @@ def to_tracking_radec(self):
     @cache_translation
     def to_altaz_begin(self):
         # Docstring will be inherited. Property defined in properties.py
-        az = self._header['AZ']
-        alt = self._header['ALT']
-        # It appears SDSS defines azimuth as increasing from South through East
-        # This translates to North through East
-        az = (-az + 180.)%360.
-        altaz = AltAz(az * u.deg, alt * u.deg,
-                      obstime=self.to_datetime_begin(), location=self.to_location())
-        self._used_these_cards('AZ', 'ALT')
-        return altaz
+        try:
+            az = self._header["AZ"]
+            alt = self._header["ALT"]
+            # It appears SDSS defines azimuth as increasing
+            # from South through East. This translates to
+            # North through East
+            az = (-az + 180.)%360.
+            altaz = AltAz(az * u.deg, alt * u.deg,
+                          obstime=self.to_datetime_begin(), location=self.to_location())
+            self._used_these_cards("AZ", "ALT")
+            return altaz
+        except Exception as e:
+            if self.to_observation_type() != "science":
+                return None  # Allow Alt/Az not to be set for calibrations
+            raise(e)
 
     @cache_translation
     def to_boresight_airmass(self):
         # Docstring will be inherited. Property defined in properties.py
         try:
-            return altaz_from_degree_headers(self, (("ALT", "AZ"),),
-                                             self.to_datetime_begin()).secz.value  # This is an estimate
+            return self.to_altaz_begin().secz.value  # This is an estimate
```

Comment:
I think you can remove the try block if you do something like:
```python
altaz = self.to_altaz_begin()
if altaz is not None:
    return altaz.secz.value
```

Reply:
OK